app/controllers/mixins/containers_common_mixin.rb

```diff
@@ -210,6 +211,14 @@ def process_check_compliance(model, ids)
     end
   end
 
+  def check_smart_roles
+    %w(SmartProxy SmartState).each do |role|
+      unless MiqServer.all.any? { |s| s.has_active_role?(role.downcase) }
```

Comment:
@gmcculloug I think that's right. I don't recall, I believe we check for available proxies in zone.